Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -636,11 +636,25 @@ private extension PurchasesOrchestrator {
            let completion = self.getAndRemovePurchaseCompletedCallback(forTransaction: storeTransaction) {
             let purchasesError = ErrorUtils.purchasesError(withSKError: error)
 
-            self.operationDispatcher.dispatchOnMainActor {
-                completion(storeTransaction,
-                           nil,
-                           purchasesError.asPublicError,
-                           purchasesError.isCancelledError)
+            let isCancelled = purchasesError.isCancelledError
+
+            if isCancelled {
+                self.customerInfoManager.customerInfo(appUserID: self.appUserID,
+                                                      fetchPolicy: .cachedOrFetched) { result in
+                    self.operationDispatcher.dispatchOnMainActor {
+                        completion(storeTransaction,
+                                   result.value,
```

Comment:
Maybe rename `result` to `customerInfo`?
```suggestion
                                                      fetchPolicy: .cachedOrFetched) { customerInfo in
                    self.operationDispatcher.dispatchOnMainActor {
                        completion(storeTransaction,
                                   customerInfo.value,
```